definitions/checks/foreman/validate_external_db_version.rb

```diff
@@ -0,0 +1,28 @@
+module Checks
+  module Foreman
+    class ValidateExternalDbVersion < ForemanMaintain::Check
+      metadata do
+        description 'Make sure server is running on required database version'
+        tags :pre_upgrade
+        label :validate_external_db_version
+        confine do
+          feature(:foreman_database) && !feature(:foreman_database).local? &&
+            !check_min_version('foreman', '2.0')
+        end
+      end
+
+      def run
+        current_db_version = feature(:foreman_database).db_version
+        fail!(db_upgrade_message(current_db_version)) if current_db_version.major < 12
+      end
+
+      def db_upgrade_message(db_version)
+        product_name = feature(:instance).product_name
+
+        "\n\n*** ERROR: Server is running on PostgreSQL #{db_version} DB.\n"\
+        "*** Newer version of #{product_name} supports only PostgreSQL version 12 or above.\n"\
```

Comment:
Sorry I didn't notice this before. There is no above in this scenario, there is only support for PostgreSQL version 12.

Reply:
Fixed! Example output: 
```

================================================================================
Make sure server is running on required database version:             [FAIL]


*** ERROR: Server is running on PostgreSQL 9.2.24 database.
*** Newer version of Satellite supports only PostgreSQL version 12.
*** Before proceeding further, you must upgrade database to PostgreSQL 12.

```